src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/CSharpCodeWriter.cs

```diff
@@ -542,42 +598,50 @@ public CSharpCodeWritingScope BuildNamespace(string name)
 
         public CSharpCodeWritingScope BuildClassDeclaration(string accessibility, string name)
         {
-            return BuildClassDeclaration(accessibility, name, Enumerable.Empty<string>());
+            return BuildClassDeclaration(accessibility, name, baseType: null);
         }
 
         public CSharpCodeWritingScope BuildClassDeclaration(string accessibility, string name, string baseType)
         {
-            return BuildClassDeclaration(accessibility, name, new string[] { baseType });
+            return BuildClassDeclaration(accessibility, name, baseType, interfaces: Enumerable.Empty<string>());
         }
 
         public CSharpCodeWritingScope BuildClassDeclaration(
             string accessibility,
             string name,
-            IEnumerable<string> baseTypes)
+            string baseType,
+            IEnumerable<string> interfaces)
         {
             Write(accessibility).Write(" class ").Write(name);
 
-            if (baseTypes.Count() > 0)
+            var hasBaseType = !string.IsNullOrEmpty(baseType);
+            var hasInterfaces = interfaces != null && interfaces.Count() > 0;
+
+            if (hasBaseType || hasInterfaces)
             {
                 Write(" : ");
-                Write(string.Join(", ", baseTypes));
+
+                if (hasBaseType)
+                {
+                    Write(baseType);
+
+                    if (hasInterfaces)
+                    {
+                        WriteParameterSeparator();
+                    }
+                }
+
+                if (hasInterfaces)
+                {
+                    Write(string.Join(", ", interfaces));
+                }
             }
 
             WriteLine();
 
             return new CSharpCodeWritingScope(this);
         }
 
-        public CSharpCodeWritingScope BuildConstructor(string name)
-        {
-            return BuildConstructor("public", name);
-        }
-
-        public CSharpCodeWritingScope BuildConstructor(string accessibility, string name)
-        {
-            return BuildConstructor(accessibility, name, Enumerable.Empty<KeyValuePair<string, string>>());
-        }
-
         public CSharpCodeWritingScope BuildConstructor(
```

Comment:
Hmmmm, this is pretty odd to have on here at all. Lets inline this functionality where it's used. I believe its only used by our ViewComponent bits.